src/stores/RoomListStore.js

```diff
@@ -527,20 +544,19 @@ class RoomListStore extends Store {
                     room, category, key, this._state.lists[key], listsClone[key], lastTimestamp);
 
                 if (!pushedEntry) {
-                    // This should rarely happen: _slotRoomIntoList has several checks which attempt
+                    // XXX: This should rarely happen: _slotRoomIntoList has several checks which attempt
```

Comment:
It's not an XXX, it's an actual feature.

Reply:
we also probably want to keep the "insert at the top to get a bug report" bits, as generating the initial room lists masks the problem.